### DISCUSSION TITLE
Open Notion URL in the Notion app

### DISCUSSION THREAD

**Discussion by joshdavenport** [johnlindquist/kit#909] [Share]
Super simple one here, if you've got a notion URL in your clipboard you can quickly whack it into this script and have it open it in your local notion app instead of the browser. If the URL you've got is a `notion://` URL then it'll open in the app by just clicking it, however, often colleagues share regular `https://` Notion URLs and I prefer to look at the resource in my local version of the app instead of having that open and also Notion tabs in the browser!

[Open open-notion-url-in-notion-app in Script Kit](https://scriptkit.com/api/new?name=open-notion-url-in-notion-app&url=https://gist.githubusercontent.com/joshdavenport/c136776ba20e01975bc8a251a4d3d58d/raw/5369240264f629bb425c12f8fb1d12e799c12981/open-notion-url-in-notion-app.ts")

```js
// Name: Open Notion URL in Notion App
// Description: Open a Notion URL in the desktio Notion App instead of  the browser
// Author: Josh Davenport-Smith

import "@johnlindquist/kit"

const url = await arg('What is the Notion URL?');

if (!url.includes('notion.so')) {
    throw new Error('Not a Notion URL');
}

await applescript(String.raw`
    tell application "Notion"
        activate
        open location "${url}"
    end tell
`);
```

**Comment by johnlindquist** [comment]
Awesome!

A couple of utils to be aware of that will help out:

```js
// Clipboard history requires "accessibility permissions"
let [latest] = await getClipboardHistory()
let { value: url } = latest
if (url.includes('notion.so')) { 
    open(url)
}
```
